orc8r/tools/ansible/roles/apt_cache/tasks/main.yml

```diff
@@ -16,9 +16,4 @@
 
 - name: Delete /var/cache/apt/archives if exists
   file: path=/var/cache/apt/archives state=absent
-
-- name: Create a local directory for apt cache if not exist
-  file: path='{{ magma_root }}/.cache/apt/{{ distribution }}' state=directory
-
-- name: Create symlink for apt cache
-  file: src='{{ magma_root }}/.cache/apt/{{ distribution }}' path=/var/cache/apt/archives state=link force=yes
+  when: preburn
```

Comment:
I think this should be `when: full_provision`, together with the two deleted statements below. The idea is that the package cache is stored in `$MAGMA_ROOT/.cache/apt/focal`. `MAGMA_ROOT` is mounted from the host folder, and thus the apt cache is preserved even if we `vagrant destroy magma`, and the next machine can reuse it. The problem was that this already happened in the preburn image, and during that time `$MAGMA_ROOT` was not yet mounted from the host. So the downloaded package files ended up in `/home/vagrant/magma/.cache/apt/archives/focal` where they contributed to the size of the VM image. When we use the VM, we mount the host `$MAGMA_ROOT` to the guest `/home/vagrant/magma` folder, and everything that was lying in `/home/vagrant/magma` before is now inaccessible.

I think what happens with the current solution is that `/var/cache/apt/archives` is deleted here but recreated once we `apt install` any packages. You could check if there are any package files lying around there in the built image. To reduce the VM image size it would be sufficient to remove those package files at the end of the build. But in order to have a persistent package cache, I think we also need to include this whole file again, but all the steps should only be run during `full_provision`.

Reply:
Removing the cache is already done in https://github.com/magma/magma/blob/49c13270e473547b1311dfd8f21056724e84aa5a/orc8r/tools/packer/scripts/third_party/chef_bento/cleanup_ubuntu.sh#L63-L64

Reply:
update: now that we do most our apt installations during preburn, we might just want to delete this persistent apt cache.

Reply:
Since this Ansible role is used in the cwag VMs I didn't delete this (yet), but removed it from both AGW VMs.